scripts/corelli/calibration/database.py

```diff
@@ -36,64 +37,165 @@ def has_valid_columns(table:TableWorkspace)->bool:
     SaveNexusProcessed(data,filename)
 
 
-def load_bank_table( bankID:int, database_path:str, table_type:str = 'calibration' ) -> TableWorkspace:
+def load_bank_table( bankID:int, database_path:str, date:str, table_type:str = 'calibration' ) -> TableWorkspace:
     """
-    Function that loads a bank calibrated TableWorkspace from a single HDF5 file
+    Function that loads the latest bank calibrated TableWorkspace from a single HDF5 file
     using corelli format:
     database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs
     :param bankID bank number that is calibrated
     :param database_path location of the corelli database (absolute or relative)
            Example: database/corelli/ for
                     database/corelli/bank001/
                     database/corelli/bank002/
+    :param date current day in YYYYMMDD format
     :param table_type 'calibration', 'mask' or 'acceptance'
     :return TableWorkspace with corresponding data
     """
-    filename:str = filename_bank_table(bankID, database_path, table_type)
+    verify_date_YYYYMMDD('load_bank_table', date)
+    filename:str = filename_bank_table(bankID, database_path, date, table_type)
     outputWS = LoadNexusProcessed(filename)
     return outputWS
 
 
+def combine_temporal_banks( database_path:str, date:str, table_type:str = 'calibration' )->WorkspaceGroup:
+    """
+    Function that combines stored banks for a certain calibration date file,
+    if not available the latest calibration date is picked up.
+    Output WorkspaceGroup can be an input to combine_spatial_banks
+    Example:
+            database/corelli/
+                bank001/table_type_corelli_bank001_20201201.nxs.h5
+                bank002/table_type_corelli_bank002_20201201.nxs.h5
+                bank003/table_type_corelli_bank003_20201201.nxs.h5
+                bank004/table_type_corelli_bank004_20200601.nxs.h5
+
+    will return a WorkspaceGroup with TableWorkspace elements from the above
+    files if date = 20201201. Since bank004 wasn't calibrated on 20201201,
+    20200601 is selected as the most recent one before 20201201.
+    :param database_path location of the corelli database (absolute or relative)
+           Example: database/corelli/ for
+                    database/corelli/bank001/
+                    database/corelli/bank002/
+    :param date pivot date in YYYYMMDD
+    :param table_type 'calibration', 'mask' or 'acceptance'
+    :return WorkspaceGroup, each component is a calibrated TableWorkspace bank
+    """
+
+    tables:WorkspaceGroup = WorkspaceGroup()
+
+    bank_dirs = list(pathlib.Path(database_path).glob('bank'+ ('[0-9]' * 3) ))
+    # filter directories
+    bank_dirs = [dir for dir in bank_dirs if dir.is_dir()]
+
+    for bank_dir in bank_dirs:
+
+        # extract files in bank calibration directory
+        files = list(pathlib.Path(bank_dir).glob(table_type + '_corelli_bank' + ('[0-9]' * 3) + '*' ) )
```

Comment:
so the name of the calibration files is of the type `calibration_corelli_bank001_20201022.nxs`?

Reply:
Yes, since these are independent files they can be moved somewhere else. It's just to be self-describing in the title.